langchain/tools/azure_cognitive_services/form_recognizer.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional
+
+from pydantic import root_validator
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForToolRun,
+    CallbackManagerForToolRun,
+)
+from langchain.tools.azure_cognitive_services.utils import detect_file_src_type
+from langchain.tools.base import BaseTool
+from langchain.utils import get_from_dict_or_env
+
+logger = logging.getLogger(__name__)
+
+
+class AzureCogsFormRecognizerTool(BaseTool):
+    """Tool that queries the Azure Cognitive Services Form Recognizer API.
+
+    In order to set this up, follow instructions at:
+    https://learn.microsoft.com/en-us/azure/applied-ai-services/form-recognizer/quickstarts/get-started-sdks-rest-api?view=form-recog-3.0.0&pivots=programming-language-python
+    """
+
+    azure_cogs_key: str  #: :meta private:
+    azure_cogs_endpoint: str  #: :meta private:
+    doc_analysis_client: Any  #: :meta private:
+
+    name = "Azure Cognitive Services Form Recognizer"
+    description = (
+        "A wrapper around Azure Cognitive Services Form Recognizer. "
+        "Useful for when you need to "
+        "extract text, tables, and key-value pairs from documents. "
+        "Input should be a url to a document."
+    )
+
+    @root_validator(pre=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and endpoint exists in environment."""
+        azure_cogs_key = get_from_dict_or_env(
+            values, "azure_cogs_key", "AZURE_COGS_KEY"
+        )
+        values["azure_cogs_key"] = azure_cogs_key
```

Comment:
do we need to actually store this? afaict could only needed for client instantiation? could do something like
```python
azure_cogs_key: str = ""
```
at line 26 and delete line 44

Reply:
Yes, you're right, we don't need to store api key here (and in other tools)

Thanks for pointing out this!